packages/framework-core/src/services/read-model-store.ts

```diff
@@ -84,6 +65,55 @@ export class ReadModelStore {
     return joinKey
   }
 
+  private async applyProjectionToReadModel(
+    readModelName: string,
+    readModelID: UUID,
+    projectionMetadata: ProjectionMetadata,
+    entity: EntityInterface
+  ): Promise<unknown> {
+    const readModel = await this.fetchReadModel(readModelName, readModelID)
+    const currentReadModelVersion: number = readModel?.boosterMetadata?.version ?? 0
+    const newReadModel = this.projectionFunction(projectionMetadata)(entity, readModel)
+
+    if (newReadModel === ReadModelAction.Delete) {
+      this.logger.debug(
+        `[ReadModelDelete#project] Deleting read model ${readModelName} with ID ${readModelID}:`,
+        readModel
+      )
+      return this.provider.readModels.delete(this.config, this.logger, readModelName, readModel)
+    } else if (newReadModel === ReadModelAction.Nothing) {
+      this.logger.debug(
+        `[ReadModelStore#project] Skipping actions for ${readModelName} with ID ${readModelID}:`,
+        newReadModel
+      )
+      return
+    }
+    // Increment the read model version in 1 before storing
+    newReadModel.boosterMetadata = {
+      ...newReadModel.boosterMetadata,
+      version: currentReadModelVersion + 1,
+    }
+    this.logger.debug(
+      `[ReadModelStore#project] Storing new version of read model ${readModelName} with ID ${readModelID}:`,
+      newReadModel
+    )
+
+    return this.provider.readModels.store(
+      this.config,
+      this.logger,
+      readModelName,
+      newReadModel,
+      currentReadModelVersion
```

Comment:
Here we tell the provider that if the read model instance we want to update _is not currently_ in the version `currentReadModelVersion`, then throw the `OptimisticConcurrencyUnexpectedVersionError`